src/digitalIdentity-backend/src/main/java/didentity/amos/digitalIdentity/security/DIUserPrincipal.java

```diff
@@ -0,0 +1,51 @@
+package didentity.amos.digitalIdentity.security;
+
+import java.util.ArrayList;
+import java.util.Collection;
+
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.security.core.authority.SimpleGrantedAuthority;
+import org.springframework.security.core.userdetails.UserDetails;
+
+import didentity.amos.digitalIdentity.model.User;
+
+public class DIUserPrincipal implements UserDetails {
+    private User user;
+    
+    public DIUserPrincipal(User user) {
+        this.user = user;
+    }
+
+    @Override
+    public Collection<? extends GrantedAuthority> getAuthorities() {
+        Collection collection = new ArrayList<SimpleGrantedAuthority>();
+        collection.add(new SimpleGrantedAuthority(user.getUserRole().getAuthority()));
+        return collection;
+    }
+
+    @Override
+    public String getPassword() {
+        return "{noop}" + user.getPassword();
```

Comment:
What is this for? :) Just for my interest

Reply:
For that one :) https://docs.spring.io/spring-security/site/docs/current/api/org/springframework/security/crypto/password/NoOpPasswordEncoder.html